web-app/src/components/DipMap.js

```diff
@@ -745,6 +745,186 @@ export default class DipMap extends React.Component {
 					this.debugCount("updateMap/hidProvince");
 				}
 			}
+
+			// Here we hardcode the "special provinces" that need to be coloured for the start of the game.
+			//TODO: does this currently support "neutral?"
+			const startingprovs = [];
+
+			startingprovs.push({
+				name: "gas",
+				favours: "France",
+				influencedby: {
+					bre: "France",
+					par: "France",
+					mar: "France",
+					spa: "Neutral",
+				},
+			});
+			startingprovs.push({
+				name: "bur",
+				favours: "France",
+				influencedby: { mar: "France", par: "France", mun: "Germany" },
+			});
+			startingprovs.push({
+				name: "bur",
+				favours: "France",
+				influencedby: {
+					mar: "France",
+					par: "France",
+					mun: "Germany",
+					bel: "Neutral",
+				}, // belgium neutral
+			});
+			startingprovs.push({
+				name: "pic",
+				favours: "France",
+				influencedby: { par: "France", bre: "France", bel: "Neutral" }, // belgium neutral
+			});
+			startingprovs.push({
+				name: "tyr",
+				favours: "Austria",
+				influencedby: {
+					mun: "Germany",
+					ven: "Italy",
+					tri: "Austria",
+					vie: "Austria",
+				},
+			});
+			startingprovs.push({
+				name: "boh",
+				favours: "Austria",
+				influencedby: { mun: "Germany", vie: "Austria" },
+			});
+			startingprovs.push({
+				name: "gal",
+				favours: "Austria",
+				influencedby: {
+					bud: "Austria",
+					vie: "Austria",
+					war: "Russia",
+					rum: "Neutral",
+				}, //and rumania
+			});
+			startingprovs.push({
+				name: "ukr",
+				favours: "Russia",
+				influencedby: {
+					war: "Russia",
+					mos: "Russia",
+					stp: "Russia",
+					rum: "Neutral",
+				},
+			});
+			startingprovs.push({
+				name: "fin",
+				favours: "Russia",
+				influencedby: { stp: "Russia", swe: "Neutral", nwy: "Neutral" },
+			});
+
+			startingprovs.push({
+				name: "ruh",
+				favours: "Germany",
+				influencedby: {
+					kie: "Germany",
+					mun: "Germany",
+					bel: "Neutral",
+					hol: "Neutral",
+				}, //and belgium and holland
+			});
+			startingprovs.push({
+				name: "sil",
+				favours: "Germany",
+				influencedby: { ber: "Germany", mun: "Germany", war: "Russia" },
+			});
+			startingprovs.push({
+				name: "pru",
+				favours: "Germany",
+				influencedby: { ber: "Germany", war: "Russia" },
+			});
+			startingprovs.push({
+				name: "arm",
+				favours: "Turkey",
+				influencedby: { ank: "Turkey", smy: "Turkey", sev: "Russia" },
+			});
+
+			//Here we check each non-SC and non-Sea territory. If all surrounding SCs are of the same power and none is "Neutral", colour them that power.
+			//Get all nodes, disqualify Sea and SC, and per node collect the edges in an array.
+			for (let prov in this.state.variant.Properties.Graph.Nodes) {
+				const node = this.state.variant.Properties.Graph.Nodes[prov];
+				if (!node.SC && node.Subs[""].Flags.Land) {
+					const borderprovs = [];
+					for (let edge in node.Subs[""].Edges) {
```

Comment:
This looks like all the places you can move *to*?

Reply:
Yes. It takes a node and collects all the (land) provinces they can move to. Then, for each of those, it will calculate the colour.

Reply:
Do you want the provinces you can move from to get there instead though? Some edges aren't bidirectional.